proposals/0384-alpenglow-migration.md

```diff
@@ -0,0 +1,283 @@
+confirmed`.
+
+Next, until a migration certificate is observed, no correct validators will
+migrate, so all correct validators will vote as normal and contribute to
+optimistic confirmation. From the correctness argument, we know if a correct
+validator casts a genesis vote, they must vote for the same Alpenglow genesis
+block.
+
+This means that eventually 82% of validators will cast a genesis vote for the
+same genesis block. Because these genesis votes are reliably delivered via
+all-to-all, some correct validator will eventually get a genesis certificate.
+
+2. Next we show that once a correct validator migrates, then all correct
+   validators will eventually migrate.
+
+There are two ways for correct validators to migrate:
+
+1. A genesis certificate
+2. An Alpenglow finalization certificate
+
+The first correct validator to migrate must have gotten a `82%` genesis
+certificate. We assume at most `19%` of the cluster is Byzantine, then at least
+this correct validator will continuosly broadcast the genesis certificate until
+they see an Alpenglow finalization certificate. 
+
+This means `82% - 19% = 63%` correct nodes will eventually receive
+the genesis certificate and will migrate to Alpenglow upon receiving the
+certificate via all-to-all broadcast which guaranteees delivery.
+
+This `63%` correct validators is then sufficient to run Alpenglow and produce
+a finalized Alpenglow block, which will induce a repair/transition from any
+other correct/lagging validators.
+
+Thus from 1. we know some correct validator must eventually migrate because
+eventually they must receive a genesis certificate, then we know from 2 that
+eventually all correct validators must migrate.
+
+### Poh Migration
+
+When switching to the first Alpenglow block, we want to deprecate Poh. This will
+be done in a few steps to mitigate the amount of code changes:
+
+1. Before the end of each Alpenglow block, set the bank tick height to
+   `bank.max_tick_height() - 1`
+2. Change tick producer to only make 1 ending tick per block, so that each bank
+   will still think it has reached `bank.max_tick_height()`. This last tick is
+   necessary to coordinate with banking stage and broadcast to properly end the
+   packing/dispersion of a block. Eliminating it is possible, but a load of
+   risky work.
+3. Change `blockstore_processor::verify_ticks()` to turn off tick verification.
+
+### Duplicate block handling
+
+On all blocks descended from the Alpenglow genesis block:
+
+1. Turn off tower duplicate block handling
+2. Turn off epoch slots
+
+## Alternatives Considered
+
+Alpenswitch where we pick fixed slot intervals `N` at which to attempt to
+optimistically migrate to Alpenglow. On failure fallback to TowerBFT we try
+again at the next slot interval. This is more painful to implement because
+of the transition back and forth between Alpenglow and TowerBFT
+
+## Impact
+
+Validators will run Alpenglow after the migration boundary
```

Comment:
might be nice to spell out what is expected as far as liveness impact here. In the happy case, I believe we expect ~1 slot and a consensus round worth of lost liveness.

Of course, malicious or offline block producers could extend this, but that's no different than normal runtime.